bug in `get_epsilon_point` for points on the cell boundary

There seems to be a bug in `get_epsilon_point` for parallel simulations. This can be demonstrated using the simple test below which compares the output from `get_epsilon_point` and `get_epsilon_grid` over a set of grid points in a 2d cell of vacuum. The result everywhere is expected to be 1.0 everywhere. `get_epsilon_point` produces the correct result for the serial case but fails in the parallel case.

```py
import unittest
import numpy as np
import meep as mp

class TestGetEpsilonGrid(unittest.TestCase):

    def setUp(self):
        resolution = 10
        self.cell_size = mp.Vector3(1.0,1.0,0)

        self.sim = mp.Simulation(resolution=resolution,
                                 cell_size=self.cell_size)

        self.sim.init_sim()


    def test_get_epsilon_grid(self):
        grid_resolution = 2
        Nx = int(grid_resolution*self.cell_size.x) + 1
        Ny = int(grid_resolution*self.cell_size.y) + 1
        xtics = np.linspace(-0.5*self.cell_size.x,0.5*self.cell_size.x,Nx)
        ytics = np.linspace(-0.5*self.cell_size.y,0.5*self.cell_size.y,Ny)

        eps_grid = self.sim.get_epsilon_grid(xtics,ytics,np.array([0]))
        for xi, xv in enumerate(xtics):
            for yi, yv in enumerate(ytics):
                eps_pt = self.sim.get_epsilon_point(mp.Vector3(xv,yv))
                print("eps:, ({:.4f}, {:.4f}), {}, {}".format(xv,yv,eps_pt,eps_grid[xi,yi]))
                self.assertAlmostEqual(np.real(eps_grid[xi,yi]), np.real(eps_pt), places=6)
                self.assertAlmostEqual(np.imag(eps_grid[xi,yi]), np.imag(eps_pt), places=6)


if __name__ == '__main__':
    unittest.main()
```

**serial simulation**
```
$ mpirun -np 1 python test.py
Using MPI version 3.1, 1 processes
-----------
Initializing structure...
time for choose_chunkdivision = 9.837e-06 s
Working in 2D dimensions.
Computational cell is 1 x 1 x 0 with resolution 10
time for set_epsilon = 0.000194357 s
-----------
eps:, (-0.5000, -0.5000), (1+0j), (1+0j)
eps:, (-0.5000, 0.0000), (1+0j), (1+0j)
eps:, (-0.5000, 0.5000), (1+0j), (1+0j)
eps:, (0.0000, -0.5000), (1+0j), (1+0j)
eps:, (0.0000, 0.0000), (1+0j), (1+0j)
eps:, (0.0000, 0.5000), (1+0j), (1+0j)
eps:, (0.5000, -0.5000), (1+0j), (1+0j)
eps:, (0.5000, 0.0000), (1+0j), (1+0j)
eps:, (0.5000, 0.5000), (1+0j), (1+0j)
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK

Elapsed run time = 0.0032 s
```

**parallel simulation**
```
$ mpirun -np 2 python test.py
Using MPI version 3.1, 2 processes
-----------
Initializing structure...
Splitting into 2 chunks by voxels
time for choose_chunkdivision = 2.9331e-05 s
Working in 2D dimensions.
Computational cell is 1 x 1 x 0 with resolution 10
time for set_epsilon = 0.000185343 s
-----------
eps:, (-0.5000, -0.5000), (0.5+0j), (1+0j)
FF
Elapsed run time = 0.0035 s

======================================================================
FAIL: test_get_epsilon_grid (__main__.TestGetEpsilonGrid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_get_epsilon_grid_bug.py", line 29, in test_get_epsilon_grid
    self.assertAlmostEqual(np.real(eps_grid[xi,yi]), np.real(eps_pt), places=6)
AssertionError: 1.0 != 0.5 within 6 places

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
```

Actually, it turns out that the *serial* case also fails whenever `get_epsilon_point` is passed a position anywhere on the cell boundary.

```py
import unittest
import numpy as np
import meep as mp

class TestGetEpsilonGrid(unittest.TestCase):

    def setUp(self):
        resolution = 10
        self.cell_size = mp.Vector3(1.0,1.0,0)

        self.sim = mp.Simulation(resolution=resolution,
                                 cell_size=self.cell_size,
	                         default_material=mp.Medium(epsilon=12.25))

        self.sim.init_sim()

    def test_get_epsilon_grid(self):
        pt = mp.Vector3(-0.5,0.5,0)
        eps_grid = self.sim.get_epsilon_grid(np.array([pt.x]),np.array([pt.y]),np.array([pt.z]))
        eps_pt = self.sim.get_epsilon_point(pt)
        print("eps:, ({},{}), {}, {}".format(pt.x,pt.y,eps_pt,eps_grid))
        self.assertAlmostEqual(np.real(eps_grid), np.real(eps_pt), places=6)
        self.assertAlmostEqual(np.imag(eps_grid), np.imag(eps_pt), places=6)


if __name__ == '__main__':
    unittest.main()

```

```
$ mpirun -np 1 python test.py
Using MPI version 3.1, 1 processes
-----------
Initializing structure...
time for choose_chunkdivision = 1.027e-05 s
Working in 2D dimensions.
Computational cell is 1 x 1 x 0 with resolution 10
time for set_epsilon = 0.00018729 s
-----------
eps:, (-0.5,0.5), (1.180722891566265+0j), (12.25+0j)
F
======================================================================
FAIL: test_get_epsilon_grid (__main__.TestGetEpsilonGrid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_get_epsilon_grid_bug.py", line 37, in test_get_epsilon_grid
    self.assertAlmostEqual(np.real(eps_grid), np.real(eps_pt), places=6)
AssertionError: array(12.25) != 1.180722891566265 within 6 places

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
```

--- a/python/tests/test_material_grid.py
+++ b/python/tests/test_material_grid.py
@@ -159,7 +159,7 @@ class TestMaterialGrid(unittest.TestCase):
     def test_symmetry(self):
             tran_nosym = compute_transmittance(False)
             tran_sym = compute_transmittance(True)
-            self.assertAlmostEqual(tran_nosym, tran_sym)
+            self.assertAlmostEqual(tran_nosym, tran_sym, places=6)
 
 if __name__ == '__main__':
     unittest.main()

--- a/src/monitor.cpp
+++ b/src/monitor.cpp
@@ -178,7 +178,7 @@ complex<double> fields::get_chi1inv(component c, direction d, const ivec &origlo
                             0);
         return parallel ? sum_to_all(val) : val;
       }
-  return d == component_direction(c) ? 1.0 : 0; // default to vacuum outside computational cell
+  return d == component_direction(c) && (parallel || am_master()) ? 1.0 : 0; // default to vacuum outside computational cell
 }
 
 complex<double> fields_chunk::get_chi1inv(component c, direction d, const ivec &iloc,

